include/igl/blue_noise.cpp

```diff
@@ -0,0 +1,390 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+// 
+// Copyright (C) 2020 Rinat Abdrashitov <rinat@dgp.toronto.edu>
+// 
+// This Source Code Form is subject to the terms of the Mozilla Public License 
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+// obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "blue_noise.h"
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <cassert>
+#include <stdint.h>
+#include <vector>
+#include <Eigen/Core>
+
+//TODO: reimplement bridsons code below using Eigen
+namespace bridson
+{
+    using std::min;
+    using std::max;
+    using std::swap;
+
+    template<class T>
+    inline T sqr(const T &x)
+    { return x*x; }
+
+    // transforms even the sequence 0,1,2,3,... into reasonably good random numbers 
+    // challenge: improve on this in speed and "randomness"!
+    inline unsigned int randhash(unsigned int seed)
+    {
+        unsigned int i=(seed^12345391u)*2654435769u;
```

Comment:
It would be preferable to use C++11 `<random>` generators directly. E.g. look at `igl::randperm`.